pkg/kubelet/network/dns/dns.go

```diff
@@ -63,20 +66,50 @@ type Configurer struct {
 	// the container's DNS resolver configuration file. This can be used in
 	// conjunction with clusterDomain and clusterDNS.
 	ResolverConfig string
+
+	// client to access network CR
+	arktosV1Client arktosv1.ArktosV1Interface
 }
 
 // NewConfigurer returns a DNS configurer for launching pods.
-func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, nodeIP net.IP, clusterDNS []net.IP, clusterDomain, resolverConfig string) *Configurer {
+func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, nodeIP net.IP, clusterDNS []net.IP, clusterDomain, resolverConfig string, arktosV1Client arktosv1.ArktosV1Interface) *Configurer {
 	return &Configurer{
 		recorder:       recorder,
 		nodeRef:        nodeRef,
 		nodeIP:         nodeIP,
 		clusterDNS:     clusterDNS,
 		ClusterDomain:  clusterDomain,
 		ResolverConfig: resolverConfig,
+		arktosV1Client: arktosV1Client,
 	}
 }
 
+func (c *Configurer) getClusterDNS(pod *v1.Pod) []net.IP {
+	const defaultNetwork = "default"
+	networkName, ok := pod.Labels["arktos.futurewei.com/network"]
+	if !ok {
+		networkName = defaultNetwork
+	}
+	if network, err := c.arktosV1Client.NetworksWithMultiTenancy(pod.Tenant).Get(networkName, metav1.GetOptions{}); err == nil {
+		sip := network.Status.DNSServiceIP
+		ip := net.ParseIP(sip)
+		if ip != nil {
+			return []net.IP{ip}
+		}
+	} else {
+		klog.Errorf("failed to get network %s/%s: %v", pod.Tenant, networkName, err)
+	}
+
+	// for now, fallback to default cluster DNS for pod that has no explicit network and no valid dns service ip of the default network not set yet.
+	// This is temporary measure to keep system unbroken before all multi-tenant network dependencies are in place.
+	// todo: disable the fallback when every components are ready.
+	if networkName == defaultNetwork {
```

Comment:
The logic seems unclear to me. Why do we need to compare network name to defaultNetwork here?  The default network is not special except that it's pre-created, which also has its own DNS service. Do we only fallback for default networks?